### PR TITLE
Check price greater than 0

### PR DESCRIPTION
**Pull request by pmenendz**
- mx-cfdi-v4: Validate price is greater than 0 as stated in the specifications:

`ValorUnitario: Si el tipo de comprobante es de “I” (Ingreso), “E” (Egreso) o “N”
(Nómina) este valor debe ser mayor a cero` 

We only support I and E

## Pre-Review Checklist

- [x] I've read the CONTRIBUTING.md guide.
- [x] I have performed a self-review of my code.
- [x] I have added thorough tests with at **least** 90% code coverage.
- [ ] I've modified or created example GOBL documents to show my changes in use, if appropriate.
- [ ] When adding or modifying a tax regime or addon, I've added links to the source of the changes, either structured or in the comments.
- [x] I've run `go generate .` to ensure that the Schemas and Regime data are up to date.
- [ ] All linter warnings have been reviewed and fixed.
- [x] I've been obsessive with pointer nil checks to avoid panics.
- [x] The CHANGELOG.md has been updated with an overview of my changes.
- [x] Requested a review from @samlown.

### CODE REVIEW

**Review by cavalle — APPROVED** [review]
LGTM!

File: addons/mx/cfdi/invoice_test.go [hunk 1]
```diff
@@ -396,6 +396,28 @@ func TestInvoiceLineItemValidation(t *testing.T) {
...
+			name: "negative price",
+			item: &org.Item{
+				Name:  "Test purchase",
+				Price: num.NewAmount(-5000, 2),
+				Ext: tax.Extensions{
+					cfdi.ExtKeyProdServ: "12345678",
+				},
+			},
+			err: "price: must be greater than 0",
+		},
```
Comment (samlown) [inline addons/mx/cfdi/invoice_test.go:420]: Another test is required here for when the price is nil.

**Review by samlown — CHANGES_REQUESTED** [review]
Nil test would be good here. It feels like this should be something more general though.

**Review by samlown — APPROVED** [review]
👍
